src/login/controller.ts

```diff
@@ -17,7 +17,7 @@ class LoginController extends Controller {
     ctx.response.body = loginForm(
       ctx.query.msg,
       ctx.query.error,
-      await getSetting('registration.enabled'),
+      (await getSetting('registration.enabled') === 'true'),
```

Comment:
hmm why is this compared to a string?

Reply:
when i changed the `registration.enabled` to false in `server-setting` table, it actually didn't recognize as a boolean, instead the type was a `string`, so i had to convert it

Reply:
I think the intent of the setting system is to treat the data as JSON, so it might mean that we're missing a `JSON.parse()` in the system that deals with getting settings from the database

Reply:
Ah okay, so since the `getSetting` function takes the string and returns a string, i think that's why `true` is returning as a string. do you think it's bad idea to wrap that `await getSetting()` function in `JSON.parse()`, since there are other places that actually uses `getSetting()` function with params as a string?

Reply:
No, I think JSON.parse should be done for all settings when they are being loaded in.

Reply:
so from `function loginForm(msg: string, error: string, registrationEnabled: boolean)`
the `registrationEnabled`'s value is a string `'true'` when i console log it. it's strange since we're explicitly setting the registrationEnabled's type as a boolean.

Reply:
getSetting() returns `any` and that basically causes typescript to completely stop type checking.

Reply:
From the `getSetting()` func, would this be an 'okay' way to do?
```
if (value == 'true' || 'false') {
  return JSON.parse(value);
}
return value;
```

Reply:
No, JSON.parse should be called for every value

Reply:
gotcha